### DISCUSSION TITLE
MVC Template for CMS Renders 404 Error

### DISCUSSION THREAD

**Discussion by krchome** [OrchardCMS/OrchardCore#16105] [Q&A]
Steps to Recreate the issue:
1) As per the Orchard Core Documentation [link](https://docs.orchardcore.net/en/latest/docs/guides/create-cms-application/) I installed the Orchard Core termplate by the CLI command: 
dotnet new install OrchardCore.ProjectTemplates::1.5.0-*
When the templates were successfully installed it showed the MVC template ocmvc having been installed as well
![image](https://user-images.githubusercontent.com/8474855/212611199-3f5420f9-81a9-498b-b42c-03741fd71ad6.png)
2) I issued the command: dotnet new ocmvc -n MySite to create an Orchard Core MVC App
3) As per the documenatation I run the project with : dotnet run --project .\MySite\MySite.csproj
5) When the project was loaded on Chrome browser, it ended up with 404 Not Found Error at : localhost:5001
![image](https://user-images.githubusercontent.com/8474855/212612313-0654a9d8-9ac2-4e2a-a71d-9e119582b78e.png)

So if it is a known bug, it may be closed with suitable reference issue.

**Comment by jtkech** [comment]
I think `ocmvc` is an empty project that would need to reference at least one module providing an home page e.g. the module built with the `ocmodulemvc` command.

For a better first experience I recommend to use `occms` instead as it already references many modules packages.

**Comment by krchome** [comment]
I know occms (Razor Pages App) is preferable and have been using it for the last couple of years in the earlier versions of OC. Now it so happens that some of my YouTube channel community members have asked me if OrchardCore CMS Website could similarly be created with MVC template . Hence I was investigating if it works equally well with MVC and found this out.
However when I took the GitHub repo project and ran the ocmvc as a startup project, it rendered a Hello World sandbox and then I guessed that the MVC part isn't developed so far at least. 

However, how can I  "reference at least one module providing an home page e.g. the module built with the ocmodulemvc command." in practice? Any help with this will be appreciated. 

Thanks very much.

**Comment by ns8482e** [comment]
Create mvc module project using `dotnet new ocmodulemvc` and add project reference to your mvc web

**Comment by krchome** [comment]
Hi your suggestion doesn't work for me, dotnet new ocmodulemvc creates just a shell of a project and when I create another project within the same solution by dotnet new ocmvc and put the project reference to this mvc project it ends up in several errors for the ocmodulemvc project StartUp.
